backend/src/controllers/user.rs

```diff
@@ -134,7 +136,14 @@ pub async fn delete_user(
         .await
         .map_err(|_| CustomError::UserNotFound(id.to_string()))?;
 
-    Ok((StatusCode::OK, Json(json!({"message": "User deleted"}))))
+    let _ = sqlx::query!(
+        "UPDATE message SET user_id = NULL WHERE user_id = $1",
+        id
+    )
+    .execute(&pool)
+    .await;
```

Comment:
This won't work, `user_id` isn't nullable.  I would just do a soft delete on the user model and update them to be `deleted=true` so you don't need to mess with cascading for now.  Then your `list` will have to filter out the deleted ones.